proposals/list-patterns.md

```diff
@@ -34,12 +34,18 @@ There are two new patterns:
 
 #### Pattern compatibility
 
-A *list_pattern* is compatible with any type that is *countable* as well as *indexable* — it has an accessible indexer that takes an `Index` as an argument or otherwise an accessible indexer with a single `int` parameter. If both indexers are present, the former is preferred.  
+A *list_pattern* is compatible with any type that is *countable* and *indexable*.
 
-A *slice_pattern* with a subpattern is compatible with any type that is *countable* as well as *sliceable* — it has an accessible indexer that takes a `Range` as an argument or otherwise an accessible `Slice` method with two `int` parameters. If both are present, the former is preferred.
+A *slice_pattern* with a subpattern is compatible with any type that is *countable* and *sliceable*.
 
 A *slice_pattern* without a subpattern is compatible with any type that is compatible with a *list_pattern*.
 
+A type is *countable* if it has an accessible property getter that returns an `int` and has the name `Length` or `Count`. If both properties are present, the former is preferred.
```

Comment:
:memo: This definition was removed in a previous PR.

Reply:
Why do we need to add it back? Both concepts already exist via the C# features around range and index, I'm not sure we should duplicate here.

Reply:
Reverted.